rpc/src/module/chain.rs

```diff
@@ -364,4 +377,121 @@ impl ChainRpc for ChainRpcImpl {
                 .map(Into::into)
         }))
     }
+
+    fn get_transaction_proof(
+        &self,
+        tx_hashes: Vec<H256>,
+        block_hash: Option<H256>,
+    ) -> Result<TransactionProof> {
+        if tx_hashes.is_empty() {
+            return Err(RPCError::invalid_params("Empty transaction hashes"));
+        }
+        let snapshot = self.shared.snapshot();
+
+        let mut retrieved_block_hash = None;
+        let mut tx_indices = HashSet::new();
+        for tx_hash in tx_hashes {
+            match snapshot.get_transaction_info(&tx_hash.pack()) {
+                Some(tx_info) => {
+                    if retrieved_block_hash.is_none() {
+                        retrieved_block_hash = Some(tx_info.block_hash);
+                    } else if Some(tx_info.block_hash) != retrieved_block_hash {
+                        return Err(RPCError::invalid_params(
+                            "Not all transactions found in retrieved block",
+                        ));
+                    }
+
+                    if !tx_indices.insert(tx_info.index as u32) {
+                        return Err(RPCError::invalid_params(format!(
+                            "Duplicated tx_hash {:#x}",
+                            tx_hash
+                        )));
+                    }
+                }
+                None => {
+                    return Err(RPCError::invalid_params(format!(
+                        "Transaction {:#x} not yet in block",
+                        tx_hash
+                    )));
+                }
+            }
+        }
+
+        let retrieved_block_hash = retrieved_block_hash.expect("checked len");
+        if let Some(specified_block_hash) = block_hash {
+            if !retrieved_block_hash.eq(&specified_block_hash.pack()) {
+                return Err(RPCError::invalid_params(
+                    "Not all transactions found in specified block",
+                ));
+            }
+        }
+
+        snapshot
+            .get_block(&retrieved_block_hash)
```

Comment:
I'm not sure whether it is a performance concern to read the whole block to only verify several leaves. As I mentioned in https://github.com/nervosnetwork/merkle-tree/pull/20/files#r484305194 , its possible to only read the transactions in the proof.